### PR TITLE
MAID-1638: refactor/examples: fix examples

### PR DESCRIPTION
**Pull request by Fraser999**
This allows the ci_test example to pass again, but with a reduced network size of 10 nodes.  It also
fixes a long-standing bug in the key_value_store example.

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
r? @afck

(maidsafe_highfive has picked a reviewer for you, use r? to override)

File: Cargo.toml [hunk 1]
```diff
@@ -27,7 +27,7 @@ unwrap = "~1.1.0"
 
 [dev-dependencies]
 docopt = "~0.6.86"
-libc = "~0.2.16"
+libc = "~0.2.17"
```
Comment (dhardy) [inline Cargo.toml]: If you did `cargo update`, shouldn't this be committed separately?

Comment (Fraser999) [inline Cargo.toml]: I just did the changes manually (I think cargo update would only touch the lock file anyway, but there is a tool for updating the toml file which some of the other guys use I think), but we don't really have an agreed-upon approach.

I generally try and keep Cargo.toml up to date whenever I remember and have been just adding any changes as part of unrelated commits.

**Comment by dirvine** [comment]
@Fraser999  @dhardy `cargo outdated` will automatically show outdated libs in your cargo.toml. Installable via `cargo install cargo-outdated`

File: examples/ci_test.rs [hunk 2]
```diff
@@ -55,33 +55,29 @@ extern crate unwrap;
...
 const CHURN_MIN_WAIT_SEC: u64 = 20;
 const CHURN_MAX_WAIT_SEC: u64 = 30;
 const CHURN_TIME_SEC: u64 = 20;
 const DEFAULT_REQUESTS: usize = 30;
-const DEFAULT_NODE_COUNT: usize = 20;
+const DEFAULT_NODE_COUNT: usize = 10;
 /// The number of churn-get cycles.
 const DEFAULT_BATCHES: usize = 1;
-const GROUP_SIZE: usize = 8;
+const GROUP_SIZE: usize = MIN_GROUP_SIZE;
```
Comment (Viv-Rajkumar) [inline examples/ci_test.rs]: why even have this const anymore and not use `MIN_GROUP_SIZE` directly?

Comment (Fraser999) [inline examples/ci_test.rs]: Good point.
